File: internal/tdd/adapters/assets/go/conformance/conformance_test.go
package conformance

import (
	"testing"

	"machinery.test/conformance/machinerycheck"
)

// TestConformanceWitnessPass proves one registered passing assertion through
// the byte-pinned helper transport under the native runner.
func TestConformanceWitnessPass(t *testing.T) {
	answer := 6 * 7
	machinerycheck.Check(t, "conformance/witness-pass", answer == 42)
}

// TestConformanceSubtestIdentity proves fixed-name deterministic subtests:
// every declared child leaf executes under its complete native path.
func TestConformanceSubtestIdentity(t *testing.T) {
	t.Run("first", func(t *testing.T) {
		machinerycheck.Check(t, "conformance/subtest-first", 1+1 == 2)
	})
	t.Run("second", func(t *testing.T) {
		machinerycheck.Check(t, "conformance/subtest-second", 2+2 == 4)
	})
}

// TestConformanceMultipleAssertions proves two registered assertions reached
// sequentially within one native test.
func TestConformanceMultipleAssertions(t *testing.T) {
	machinerycheck.Check(t, "conformance/multi-a", len("machinery") == 9)
	machinerycheck.Check(t, "conformance/multi-b", len("assurance") == 9)
}

File: internal/tdd/adapters/assets/elixir/conformance/conformance_test.exs
defmodule Machinery.ConformanceTest do
  @moduledoc false
  use ExUnit.Case, async: true

  require Machinery.Check

  test "conformance witness executes native assertion", ctx do
    Machinery.Check.check(ctx, "conformance/witness", 6 * 7 == 42)
  end

  describe "conformance parent identity" do
    test "conformance nested identity", ctx do
      Machinery.Check.check(ctx, "conformance/nested", 2 * 7 == 14)
    end
  end

  test "conformance multiple assertions", ctx do
    Machinery.Check.check(ctx, "conformance/multi-a", 3 * 5 == 15)
    Machinery.Check.check(ctx, "conformance/multi-b", 4 * 5 == 20)
  end
end
